cli/crates/cli/src/introspect.rs

```diff
@@ -28,11 +28,52 @@ fn introspect_local() -> Result<(), CliError> {
     Ok(())
 }
 
-fn introspect_remote(url: &str, headers: &[(&str, &str)]) -> Result<(), CliError> {
+fn introspect_remote(url: &str, headers: &[(&str, &str)], no_color: bool) -> Result<(), CliError> {
     let operation = graphql_introspection::introspect(url, headers);
 
     match Runtime::new().unwrap().block_on(operation) {
-        Ok(result) => Ok(println!("{result}")),
+        Ok(result) => {
+            print_introspected_schema(&result, no_color);
+            Ok(())
+        }
         Err(e) => Err(CliError::Introspection(e)),
     }
 }
+
+fn print_introspected_schema(sdl: &str, no_color: bool) {
+    use syntect::util::{as_24_bit_terminal_escaped, LinesWithEndings};
+
+    // No highlighting when stdout is not a tty (likely a pipe) or when explicitly requested.
+    if no_color || !atty::is(atty::Stream::Stdout) || no_color_env_var() {
+        println!("{sdl}");
+        return;
+    }
+
+    const GRAMMAR: &str = include_str!("introspect/graphql_grammar.yml");
+
+    let mut builder = syntect::parsing::SyntaxSetBuilder::new();
+    builder.add(syntect::parsing::SyntaxDefinition::load_from_str(GRAMMAR, false, Some("graphql")).unwrap());
```

Comment:
Could we make this an expect or a user facing error?

Reply:
Addressed in commit (with expect, since there's really no variability, it always loads exactly the same grammar definition baked into the binary).